docs/platform/tutorials/eventDetection/dataAcquisition.md

```diff
@@ -0,0 +1,283 @@
+
+## CSV data
+
+If you don’t have an Android device or you’d rather just stream some data we’ve provided then this is the data source for you! 
+
+Follow these instructions to deploy the data source:
+
+1. In the Quix Library, select `Python` under languages and `Source` under pipeline stage
+
+2. In the search box enter `Empty template`
+
+3. On `Empty template - Source`, click `Preview code` then `Edit code`
+
+4. Change the `Name` field to `CSV data source`
+
+5. Change the `output` field to `phone-data`
+
+6. Click `Save as project`
+
+7. Open the requirements.txt file and add `urllib3` to a new line
+
+8. Open the `main.py` file and replace the code with the following code:
+
+	```py
+	from quixstreaming import QuixStreamingClient
+	from quixstreaming.app import App
+	import time
+	import pandas as pd
+	import os
+	from urllib import request
+
+
+	# download data.csv
+	# if you want to use your own data file just comment these lines
+	# and place your data.csv file in the same folder as main.py
+	f = request.urlopen("https://quixtutorials.blob.core.windows.net/tutorials/event-detection/data.csv")
+	with open("data.csv", "wb") as data_file:
+		data_file.write(f.read())
+
+	# Quix injects credentials automatically to the client. Alternatively, you can always pass an SDK token manually as an argument.
+	client = QuixStreamingClient()
+
+	# Open the output topic
+	print("Opening output topic")
+	output_topic = client.open_output_topic(os.environ["output"])
+
+	output_stream = output_topic.create_stream()
+
+	df = pd.read_csv("data.csv")
+
+	for col_i in ['device_id','rider','streamId','team','version']:
+		df = df.rename(columns={col_i: "TAG__" + col_i})
+		
+	print("Writing data")
+	seconds_to_wait = 0.5
+
+	while True:
+		for i in range(len(df)):
+			start_loop = time.time()
+
+			df_i = df.iloc[[i]]
+
+			output_stream.parameters.write(df_i)
+			print("Sending " + str(i) + "/" + str(len(df)))
+			end_loop = time.time()
+			time.sleep(max(0.0, seconds_to_wait - (end_loop - start_loop)))
+
+	App.run()
+
+	```
+
+	!!! info 
+		The code:
+
+		- Connects to Quix
+
+		- Opens the CSV file using Pandas
+
+		- Renames some columns so the Quix SDK teams them as tags
+
+		- Then streams each row to the `phone-data` topic
+
+		- It does this continuously so stop the service when you have completed the tutorial
+
+7. Test the code by clicking `run` near the top right corner of the code window
+
+	With the code running you will see messages in the `Console` tab
+
+	```sh
+	[xx-xx-xx xx:xx:xx.xxx (8) INF] Topic phone-data is still creating
+	Opening output topic
+	Writing data
+	Sending 0/18188
+	Sending 1/18188
+	Sending 2/18188
+	```
+
+8. Click the `Messages` tab and you’ll see the raw messages being streamed to the `phone-data` topic.
+
+9. Click one of the messages and you’ll see the raw data in [JSON](https://www.w3schools.com/whatis/whatis_json.asp){target=_blank} format
+
+	???- info "Your data should look like this"
```

Comment:
Mine didn't look like that. On investigation it turns out that the messages can have a different format - for example, some have geolocation data and some don't. We should just add a note that the message format can vary one one message to the next.